datafusion/src/execution/context.rs

```diff
@@ -846,39 +773,14 @@ pub struct SessionConfig {
     /// parallel using the provided `target_partitions` level
     pub repartition_windows: bool,
     /// Should DataFusion parquet reader using the predicate to prune data
-    parquet_pruning: bool,
-    /// Runtime configurations such as memory threshold and local disk for spill
-    pub runtime: RuntimeConfig,
+    pub parquet_pruning: bool,
```

Comment:
@yjshen I agree that the interface with getters would be good;

```rust
pub fn get_bool(&self, key: &str) -> Result<bool> {
    self.get(key).and_then(Value::into_bool)
}
```

Among other things that would allow us to change the implementation (e.g. to properties or typed fields) without issue 

I do think typed fields have some nice properties (like being able to do possibly earlier error detection) but in order to support extensions/customized properties we definitely need to support aribtrary key/values as well

In terms of `TaskProperties` I personally prefer something like this:

```rust
pub struct TaskProperties {
    ///SessionConfig
    config: SessionConfig,
    /// Additional Name-value pairs of task properties
    properties: HashMap<String, String>,
}
```

Reply:
How about this one, to avoid having a SessionConfig and a HashMap.

```rust

#[derive(Clone, Debug)]
enum Value {
    USIZE(usize),
    STRING(String),
    BOOL(bool),
}

impl Value {
    fn into_usize(&self) -> Result<usize> {
        if let Value::USIZE(u) = self {
            Ok(*u)
        } else {
            Err(DataFusionError::Internal(format!("{:?} not a usize conf", self)))
        }
    }

    fn into_string(&self) -> Result<String> {
        if let Value::STRING(s) = self {
            Ok(s.to_owned())
        } else {
            Err(DataFusionError::Internal(format!("{:?} not a string conf", self)))
        }
    }
}

struct TaskProperties {
    pub inner: HashMap<String, Value>,
}

impl TaskProperties {
    /// setters
    fn set_usize(&mut self, key: impl Into<String>, value: usize) {
        self.inner.insert(key.into(), Value::USIZE(value));
    }

    fn set_bool(&mut self, key: impl Into<String>, value: bool) {
        self.inner.insert(key.into(), Value::BOOL(value));
    }

    /// getters
    fn get_usize(&self, key: &str, default: usize) -> usize {
        self.inner.get(key).and_then(|x| x.into_usize().ok()).unwrap_or(default)
    }

    /// known conf fast passes
    fn batch_size(&self) -> usize {
        self.get_usize("target_batch_size", 10240)
    }
}
```

Reply:
> How about this one, to avoid having a SessionConfig and a HashMap.

I think it is overkill at this point, to be honest. If we want to have dynamically typed versions, perhaps we can use `ScalarValue`. 

Starting with a single hash map with strings, and well documented getter/setter functions would be fine  and we can optimizer performance / error handling later

Reply:
> a single hash map with strings

Make sense to me.

Reply:
> > How about this one, to avoid having a SessionConfig and a HashMap.
> 
> I think it is overkill at this point, to be honest. If we want to have dynamically typed versions, perhaps we can use `ScalarValue`.
> 
> Starting with a single hash map with strings, and well documented getter/setter functions would be fine and we can optimizer performance / error handling later

I prefer to use a hash map or other more extendable structs too. But can I leave this change in another PR ?